mobx/lib/src/api/observable_collections/observable_list.dart

```diff
@@ -60,8 +60,15 @@ class ObservableList<T>
   set length(int value) {
     /// There is no need to enforceWritePolicy since we are conditionally wrapping in an Action.
     _context.conditionallyRunInAction(() {
-      _list.length = value;
-      _notifyListUpdate(0, null, null);
+      if (value < _list.length) {
+        final removed = _list.sublist(value);
+        _list.length = value;
+        _notifyRangeUpdate(value, null, removed);
+      } else if (value > _list.length) {
+        final index = _list.length;
+        _list.length = value;
+        _notifyRangeUpdate(index, _list.sublist(index), null);
+      }
```

Comment:
This was a big miss since the usage was very less. Thanks for pitching in and fixing this part 👍